db/revision.go

```diff
@@ -364,14 +364,18 @@ func (body Body) FixJSONNumbers() {
 
 func createRevID(generation int, parentRevID string, body Body) (string, error) {
 	// This should produce the same results as TouchDB.
-	digester := md5.New()
-	digester.Write([]byte{byte(len(parentRevID))})
-	digester.Write([]byte(parentRevID))
 	encoding, err := base.JSONMarshalCanonical(stripSpecialProperties(body))
 	if err != nil {
 		return "", err
 	}
-	digester.Write(encoding)
+	return CreateRevIDWithBytes(generation, parentRevID, encoding)
+}
+
+func CreateRevIDWithBytes(generation int, parentRevID string, bodyBytes []byte) (string, error) {
```

Comment:
This doesn't need to return error, unless you were planning to add handling for the unhandled error on digester.Write.